src/components/SnapshotImage.tsx

```diff
@@ -106,47 +141,87 @@ export const SnapshotImage = ({
   const showDiff = hasDiff && diffVisible;
   const showFocus = hasFocus && focusVisible;
 
+  const [latestImageLoaded, setLatestImageLoaded] = React.useState(false);
+  const [baselineImageLoaded, setBaselineImageLoaded] = React.useState(false);
+  const [comparisonImageLoaded, setComparisonImageLoaded] = React.useState(false);
+  const [focusImageLoaded, setFocusImageLoaded] = React.useState(false);
+  const snapshotImageLoaded = baselineImageVisible ? baselineImageLoaded : latestImageLoaded;
+
+  let overlayImageLoaded = true;
+  if (showDiff && showFocus) {
+    overlayImageLoaded = comparisonImageLoaded && focusImageLoaded;
+  } else if (showDiff) {
+    overlayImageLoaded = comparisonImageLoaded;
+  } else if (showFocus) {
+    overlayImageLoaded = focusImageLoaded;
+  }
```

Comment:
I would prefer to use a `useMemo` or a simple `getOverlayImageLoaded` function here instead, which makes the code more readable and interrelated.